eis_toolkit/utilities/nodata.py

```diff
@@ -0,0 +1,119 @@
+
+    Returns:
+        out_raster_data: The original raster data with replaced nodata values.
+
+    Raises:
+        InvalidRasterBandException: Invalid band index in nodata mapping.
+    """
+    if any(band > len(raster_data) or band < 1 for band in nodata_per_band.keys()):
+        raise InvalidRasterBandException("Invalid band index in nodata mapping.")
+
+    out_raster_data = raster_data.copy()
+
+    for band, nodata_values in nodata_per_band.items():
+        index = band - 1
+        band_data = raster_data[index]
+        out_data = replace_values_with_nodata(band_data, nodata_values, new_nodata)
+        out_raster_data[index] = out_data
+
+    return out_raster_data
+
+
+def replace_values_with_nodata(
+    data: np.ndarray, values_to_replace: List[float], new_nodata: float = np.nan
+) -> np.ndarray:
+    """
+    Replace multiple nodata values in a raster numpy array with a new nodata value.
+
+    Args:
+        in_data: Input raster data as a numpy array.
+        values_to_replace: List of values to be replaced with new_nodata.
+        new_nodata: New nodata value to be set. Defaults to np.nan.
+
+    Returns:
+        out_data: Raster data with updated nodata values.
+    """
+    out_data = data.copy()
+    out_data[np.isinf(out_data)] = new_nodata  # Is this line needed?
+    mask = np.isin(data, values_to_replace)
+    out_data[mask] = new_nodata
+    return out_data
+
+
+def replace_nodata_dataframe(
+    df: pd.DataFrame, old_nodata: float | List[float], new_nodata: float = np.nan, columns: Optional[List[str]] = None
+) -> pd.DataFrame:
+    """
+    Replace the nodata value in specified columns of a DataFrame.
+
+    Args:
+        df: Input DataFrame data.
+        old_nodata: Current nodata value(s) to be replaced.
+        new_nodata: New nodata value to be set. Defaults to np.nan.
+        columns: List of column names to replace nodata values. Defaults to None (all columns).
+
+    Returns:
+        out_df: DataFrame with updated nodata values.
+    """
+    if columns is None:
+        columns = df.columns
+    else:
+        if not check_columns_valid(df, columns):
+            raise InvalidColumnException("All columns were not found in input dataframe.")
+
+    out_df = df.copy()
+
+    for col in columns:
+        out_df[col] = out_df[col].replace(old_nodata, new_nodata)
+
+    return out_df
+
+
+def handle_nodata_as_nan(func: Callable):
+    """Replace nodata_values with np.nan for function execution and reverses the replacement afterwards."""
+
+    @functools.wraps(func)
+    def wrapper(in_data: np.ndarray, *args: Any, nodata_values: List[np.number], **kwargs: Any) -> np.ndarray:
+        replaced_data = replace_values_with_nodata(in_data, nodata_values, np.nan)
+        result = func(replaced_data, *args, **kwargs)
+        out_data = replace_values_with_nodata(result, np.nan, nodata_values)
+        return out_data
+
+    return wrapper
```

Comment:
I think all downstream functions can probably expect that the nodata is already handled (i.e. replaced with `np.nan`). I cannot readily say when we would have functions that require the explicit value of nodata from the raster instead of `np.nan`. So this decorator would mostly be in use close to the command-line so the implementation details are probably up to the person who does the command-line.